modules/ocs2/src/main/scala/gem/ocs2/pio/PioDecoder.scala

```diff
@@ -23,16 +24,16 @@ object PioDecoder {
         f(n)
     }
 
-  def enum[A](dataType: String)(m: (String, A)*): PioDecoder[A] =
-    fromParseFunction[A](dataType) { m.toMap.lift }
+  def enum[A: ClassTag](m: (String, A)*): PioDecoder[A] =
```

Comment:
I just learned about this ... I think a better way to do this is with `shapeless.Typeable` which doesn't use the reflection API at runtime so it will work in scala-js and isn't prone to threading issues. It also does the name shortening for you, which can actually fail at runtime because Scala uses names that violate the JVM spec in some places.

```
scala> def foo[A](implicit ev: shapeless.Typeable[A]) = s"the type was ${ev.describe}"
foo: [A](implicit ev: shapeless.Typeable[A])String

scala> foo[gem.Dataset.Label]
res0: String = the type was Label
```